proof/crefine/ARM_HYP/Wellformed_C.thy

```diff
@@ -532,6 +532,51 @@ lemma ucast_irq_array_guard[unfolded irq_array_size_val, simplified]:
 
 text \<open>cacheLineBits interface\<close>
 
+(* only use this inside cache op functions; see Arch_Kernel_Config_Lemmas.cacheLineBits_sanity *)
+lemmas cacheLineBits_val =
+  cacheLineBits_def[unfolded Kernel_Config.CONFIG_L1_CACHE_LINE_SIZE_BITS_def]
+
+lemma cacheLineBits_le_ptBits:
+  "cacheLineBits \<le> ptBits"
+  using cacheLineBits_sanity
+  by (simp add: pt_bits_def pte_bits_def)
+
+lemma ptBits_leq_pageBits:
+  "ptBits \<le> pageBits"
+  by (simp add: pt_bits_def pte_bits_def pageBits_def)
+
+lemma ptBits_leq_pdBits:
+  "ptBits \<le> pdBits"
+  by (simp add: pt_bits_def pd_bits_def pde_bits_def pte_bits_def)
```

Comment:
it's odd to me that the above two lemmas are in the cacheLineBits interface section

Reply:
Have added a comment for it.